internal/api/v1/services/instance.go

```diff
@@ -114,62 +119,115 @@ func (s *Instance) provisionInstances(ctx context.Context, jobID uint, instances
 			if strings.Contains(err.Error(), "404") &&
 				strings.Contains(err.Error(), "could not be found") {
-				fmt.Printf("⚠️ Warning: Some old resources were not found (already deleted)\n")
+				logger.Warn("⚠️ Warning: Some old resources were not found (already deleted)")
 			} else {
-				fmt.Printf("❌ Failed to execute infrastructure: %v\n", err)
+				logger.ErrorWithFields("Failed to execute infrastructure", map[string]interface{}{
+					"job_id": jobID,
+					"error":  err,
+				})
 				return
 			}
 		}
 
 		// Update instance information in database
 		pInstances, ok := result.([]infrastructure.InstanceInfo)
 		if !ok {
-			fmt.Printf("❌ Invalid result type: %T\n", result)
+			logger.ErrorWithFields("Invalid result type", map[string]interface{}{
+				"job_id":      jobID,
+				"result_type": fmt.Sprintf("%T", result),
+			})
 			return
 		}
 
-		fmt.Printf("📝 Created instances: %+v\n", pInstances)
+		logger.InfoWithFields("Created instances", map[string]interface{}{
+			"job_id":    jobID,
+			"instances": pInstances,
+		})
 
 		// Update instance information in database
 		for _, instance := range pInstances {
 			// Update IP and status
 			if err := s.repo.UpdateIPByName(ctx, instance.Name, instance.IP); err != nil {
-				fmt.Printf("❌ Failed to update instance %s IP: %v\n", instance.Name, err)
+				logger.ErrorWithFields("Failed to update instance IP", map[string]interface{}{
+					"instance_name": instance.Name,
+					"ip":            instance.IP,
+					"error":         err,
+				})
 				continue
 			}
-			fmt.Printf("✅ Updated instance %s with IP %s\n", instance.Name, instance.IP)
+			logger.InfoWithFields("Updated instance IP", map[string]interface{}{
+				"instance_name": instance.Name,
+				"ip":            instance.IP,
+			})
 
 			if err := s.repo.UpdateStatusByName(ctx, instance.Name, models.InstanceStatusReady); err != nil {
-				fmt.Printf("❌ Failed to update instance %s status: %v\n", instance.Name, err)
+				logger.ErrorWithFields("Failed to update instance status", map[string]interface{}{
+					"instance_name": instance.Name,
+					"status":        "ready",
+					"error":         err,
+				})
 				continue
 			}
-			fmt.Printf("✅ Updated instance %s status to ready\n", instance.Name)
+			logger.InfoWithFields("Updated instance status", map[string]interface{}{
+				"instance_name": instance.Name,
+				"status":        "ready",
+			})
 		}
 
 		// Start Ansible provisioning if requested
 		if instances[0].Provision {
 			if err := infra.RunProvisioning(pInstances); err != nil {
-				fmt.Printf("❌ Failed to run provisioning: %v\n", err)
+				logger.ErrorWithFields("Failed to run provisioning", map[string]interface{}{
+					"job_id": jobID,
+					"error":  err,
+				})
 				return
 			}
+
+			// Update IsProvisioned status for all instances after successful provisioning
+			for _, instance := range pInstances {
+				if err := s.repo.UpdateByName(ctx, instance.Name, map[string]interface{}{
+					"is_provisioned": true,
```

Comment:
would be much better if we use a special repo function just to update that field or if we want to use generic update, extract the field name into a defined constant in the models.